skrub/_sentence_encoder.py

```diff
@@ -0,0 +1,300 @@
+import numbers
+import warnings
+from pathlib import Path
+
+from sklearn.base import TransformerMixin
+from sklearn.decomposition import PCA
+from sklearn.utils.validation import check_is_fitted
+
+from . import _dataframe as sbd
+from ._on_each_column import RejectColumn, SingleColumnTransformer
+from ._utils import import_optional_dependency, unique_strings
+from .datasets._utils import get_data_dir
+
+
+class ModelNotFound(ValueError):
+    pass
+
+
+class SentenceEncoder(SingleColumnTransformer, TransformerMixin):
+    """Encode string features by applying an embedding model downloaded \
+        from the HuggingFace Hub.
+
+    This is a thin wrapper around :class:`~sentence_transformers.SentenceTransformer`
+    that follows the scikit-learn API, making it usable within a scikit-learn pipeline.
+
+    .. warning::
+
+        To use this class, you need to install the optional ``transformers``
+        dependencies for skrub. For example, you can use pip:
+
+        .. code::
+
+           pip install skrub[transformers] -U
+
+        This will install the ``sentence-transformers``, ``transformers``,
+        and ``torch`` libraries on your machine. Be aware that this might lead
+        to dependency conflicts.
+
+    This class uses a pre-trained model, so calling ``fit`` or ``fit_transform``
+    will not train or fine-tune the model. Instead, the model is loaded from disk,
+    and a PCA is fitted if ``n_components`` is not None.
+
+    When PCA is disabled, this class is essentially stateless, with loading the
+    pre-trained model from disk being the only difference between ``fit_transform``
+    and ``transform``.
+
+    Be aware that parallelizing this class (e.g., using
+    :class:`~skrub.TableVectorizer` with ``n_jobs`` > 1) may be computationally
+    expensive.
+    This is because a copy of the pre-trained model is loaded into memory
+    for each thread. If memory usage is a concern, check the characteristics of
+    your selected model.
```

Comment:
maybe set n_jobs to 1 and use the transformer library's own mechanism for parallelization within the language model?

Reply:
This class has no `n_jobs` parameter because we let pytorch set the default number of threads.

Reply:
yes sorry I didn't mean in the class but in the doc's recommendation we could say something like "set n_jobs to 1 on the TableVectorizer and let pytorch handle parallelism" . but maybe that's going too much into details

Reply:
I added this suggestion